app/controllers/bookings_controller.rb

```diff
@@ -1,16 +1,56 @@
 class BookingsController < ApplicationController
+  before_action :set_booking, only: [:show, :edit, :update]
+
   def show
+    @service = Service.find(@booking.service_id)
+    @category = Category.find(@service.category_id)
+    @user = User.find(@service.user_id)
   end
 
   def new
+    @service = Service.find(params[:service_id])
+    @booking = Booking.new
+    if user_signed_in?
+    else
+      redirect_to new_user_session_path
+    end
+    @booking.service_id = params[:service_id]
   end
 
   def create
+    @booking = Booking.new(booking_params)
+    @booking.user_id = current_user.id
```

Comment:
Here the program is gonna crash if the form doesn't pass the valdiations, because your program doesn't know @service.

So, cleaner way to write would be:

@booking = Booking.new(booking_params)
@service = Service.find(params[:service.id])
@booking.service = @service
@booking.user = current_user

lines 25 to 30 are ok :)

Reply:
ah sorry, forget:

@booking.total_price = @booking.duration * @service.price_per_hour